view/frontend/web/js/view/catalog/product/product-protection-offer.js

```diff
@@ -64,5 +65,29 @@ define(['jquery', 'extendSdk', 'ExtendMagento'], function ($, Extend, ExtendMage
         )
       }
     })
+    // Listen for the add to cart button to be clicked.  Show modal offer on qualifying simple and configurable products if no offer was chosen by the customer.
+    document.getElementById('product-addtocart-button').addEventListener('click', function () {
+      let selectedProduct
+      const buttonInstance = Extend.buttons.instance(
```

Comment:
Extend.buttons.instance can return null, should we add a check to make sure it exists before accessing getPlanSelection?